storage/libradb/src/lib.rs

```diff
@@ -420,16 +420,22 @@ impl LibraDB {
         zip_eq(first_version..=last_version, txns_to_commit)
             .map(|(ver, txn_to_commit)| {
                 self.transaction_store
-                    .put_transaction(ver, txn_to_commit.signed_txn(), &mut cs)
+                    .put_transaction(ver, txn_to_commit.transaction(), &mut cs)
             })
             .collect::<Result<()>>()?;
 
         // Transaction accumulator updates. Get result root hash.
         let txn_infos = izip!(txns_to_commit, state_root_hashes, event_root_hashes)
             .map(|(t, s, e)| {
-                TransactionInfo::new(t.signed_txn().hash(), s, e, t.gas_used(), t.major_status())
+                Ok(TransactionInfo::new(
+                    t.as_signed_user_txn()?.hash(),
```

Comment:
Maybe add a note/todo here to change this later.

Reply:
We won't forget, because later we need to change how hash is calculated on both writing and reading paths to make tests pass.